src/RawCodeCoverageData.php

```diff
@@ -21,19 +21,27 @@ final class RawCodeCoverageData
      */
     private $lineCoverage = [];
 
-    public function __construct(array $rawCoverage = [])
+    public static function fromXdebugWithoutPathCoverage(array $rawCoverage): self
     {
-        foreach ($rawCoverage as $file => $fileCoverageData) {
-            $hasOnlyIntegerKeys = \count(\array_filter(\array_keys($fileCoverageData), 'is_int')) === \count($fileCoverageData);
+        $coverage               = new self();
+        $coverage->lineCoverage = $rawCoverage;
+
+        return $coverage;
+    }
+
+    public static function fromXdebugWithPathCoverage(array $rawCoverage): self
+    {
+        $coverage = new self();
 
-            if ($hasOnlyIntegerKeys) {
-                $this->lineCoverage[$file] = $fileCoverageData;
-            } elseif (\count($fileCoverageData) === 2 && isset($fileCoverageData['lines'], $fileCoverageData['functions'])) {
-                $this->lineCoverage[$file] = $fileCoverageData['lines'];
-            } else {
-                throw UnknownCoverageDataFormatException::create($file);
-            }
+        foreach ($rawCoverage as $file => $fileCoverageData) {
+            $coverage->lineCoverage[$file] = $fileCoverageData['lines'];
         }
+
+        return $coverage;
+    }
+
+    private function __construct()
```

Comment:
Thank you formaking the constructor private and for adopting the concept of named constructors. However, the named constructor must not first create an object and then assign data to it. Instead, the named constructor should pass its parameter, maybe after mapping or processing it, to the constructor.

Reply:
Done 🤞